lib/ProductOpener/CRM.pm

```diff
@@ -0,0 +1,257 @@
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU Affero General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+
+=encoding UTF-8
+
+=head1 NAME
+
+ProductOpener::CRM - manages integration with the Odoo CRM
+
+=head1 SYNOPSIS
+
+C<ProductOpener::CRM> contains functions to interact with the Odoo CRM
+
+
+=head1 DESCRIPTION
+
+[..]
+
+=cut
+
+package ProductOpener::CRM;
+
+use ProductOpener::PerlStandards;
+use Exporter qw< import >;
+
+BEGIN {
+	use vars qw(@ISA @EXPORT_OK %EXPORT_TAGS);
+	@EXPORT_OK = qw(
+		&create_company
+		&create_contact
+		&add_contact_to_company
+		&create_opportunity
+		&test
+	);
+	%EXPORT_TAGS = (all => [@EXPORT_OK]);
+
+}
+use vars @EXPORT_OK;
+
+use ProductOpener::Tags qw(%country_codes_reverse);
+use XML::RPC;
+use Log::Any qw($log);
+use Data::Dumper;
+use feature 'say';
+
+my %odoo_tags = (
+	'Producter' => '10',
+);
+
+our @api_credentials;
+our $xmlrpc;
+
+
+=head1 FUNCTIONS
+
+=head2 create_company ($org_ref)
+
+Creates a new company in odoo from an org
+
+=head3 Arguments
+
+=head4 $org_ref
+
+=head3 Return values
+
+the id of the created company
+
+=cut
+
+sub create_company ($org_ref) {
+	my $company = {
+		name => 		$org_ref->{name}, 
+		x_off_org_id => $org_ref->{org_id}, 
+		phone => 		$org_ref->{phone},
+		email => 		$org_ref->{email}, 
+		website => 		$org_ref->{website}, 
+		category_id =>  [$odoo_tags{Producter}],  # "Producter" category id in odoo
+		is_company => 	1,
+	};
+	my $company_id = odoo('res.partner', 'create', [{%$company}]);
+	$log->debug("create_company", {company_id => $company_id}) if $log->is_debug();
+	return $company_id;
+}
+
+
+
+=head2 create_contact ($user_ref)
+
+Creates a new contact in odoo from a user
+
+=head3 Arguments
+
+=head4 $org_ref
+
+=head3 Return values
+
+the id of the created contact
+
+=cut
+
+sub create_contact ($user_ref) {
+	my $contact = {
+		name => 			$user_ref->{name}, 
+		x_off_username => 	$user_ref->{userid},
+		email => 			$user_ref->{email},
+		phone => 			$user_ref->{phone}, 
+		category_id => 		[$odoo_tags{Producter}],
+	};
+	
+	# find country code id in odoo
+	my $user_country_code = uc $country_codes_reverse{$user_ref->{country}} || 'EN';
+	my $country_id = odoo('res.country', 'search_read', [[['code', '=', $user_country_code]]], { 'fields' => ['code', 'id']});
+	$contact->{country_id} = $country_id->[0]{id};
+
+	# find spoken language's code in Odoo
+	my $odoo_lang = odoo('res.lang', 'search_read', [[['iso_code', '=', $user_ref->{preferred_language}]]], { 'fields' => ['code']});
+	my $found_lang = $odoo_lang->[0];
+	$contact->{lang} = $found_lang->{code} // 'en_US'; # default to english
+	if(defined $found_lang->{id}) {
+		$contact->{x_off_languages} = [$found_lang->{id}];
+	}
```

Comment:
There is a problem with Odoo because it will refuse to display a non-activated language in many to many fields (that's deep in the code), that's why I also created the [x_off_languages_str](https://github.com/openfoodfacts/odoo-openfoodfacts/blob/main/skill-pool/ir_model_fields.yml#L76) field to store language values directly